### PR TITLE
refactor(blockifier_reexecution): set chain ID by inspecting RPC URL

### PR DESCRIPTION
**Pull request by dorimedini-starkware**

### CODE REVIEW

**Comment by dorimedini-starkware** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/starkware-libs/sequencer/1973?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#1974** <a href="https://app.graphite.dev/github/pr/starkware-libs/sequencer/1974?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1973** <a href="https://app.graphite.dev/github/pr/starkware-libs/sequencer/1973?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#1972** <a href="https://app.graphite.dev/github/pr/starkware-libs/sequencer/1972?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1971** <a href="https://app.graphite.dev/github/pr/starkware-libs/sequencer/1971?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1970** <a href="https://app.graphite.dev/github/pr/starkware-libs/sequencer/1970?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @dorimedini-starkware and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: crates/blockifier_reexecution/src/state_reader/utils.rs [hunk 1]
```diff
@@ -19,11 +19,24 @@ use crate::state_reader::errors::ReexecutionError;
...
+pub fn guess_chain_id_from_node_url(node_url: &str) -> ReexecutionResult<ChainId> {
+    if node_url.contains("mainnet") {
+        Ok(ChainId::Mainnet)
+    } else if node_url.contains("sepolia") && !node_url.contains("integration") {
+        Ok(ChainId::Sepolia)
+    } else if node_url.contains("integration") {
+        Ok(ChainId::IntegrationSepolia)
+    } else {
+        Err(ReexecutionError::AmbiguousChainIdFromUrl(node_url.to_string()))
+    }
```
[the remark this answers is not in the document]
Comment (dorimedini-starkware) [reply crates/blockifier_reexecution/src/state_reader/utils.rs]: Done.

**Review by dorimedini-starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1973)* status: 0 of 4 files reviewed, 1 unresolved discussion (waiting on @aner-starkware and @graphite-app[bot])

**Review by AvivYossef-starkware — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1973#-:-OB_ns157zqPIBd4FfHE:bnfp4nl)* status: 0 of 4 files reviewed, 1 unresolved discussion (waiting on @aner-starkware and @graphite-app[bot])

**Review by aner-starkware — APPROVED** [review]
> Reviewed 1 of 4 files at r1, 3 of 3 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1973)* status: all files reviewed, 1 unresolved discussion (waiting on @graphite-app[bot])

**Review by dorimedini-starkware — COMMENTED** [review]
> Dismissed @graphite-app[bot] from a discussion.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1973)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @dorimedini-starkware)

**Comment by dorimedini-starkware** [comment]
### Merge activity

* **Nov 19, 10:11 AM EST**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/starkware-libs/sequencer/1973).

**Pull request merged**